src/gui/shareusergroupdialog.cpp

```diff
@@ -0,0 +1,226 @@
+ * for more details.
+ */
+
+#include "shareusergroupdialog.h"
+#include "ui_shareusergroupdialog.h"
+#include "ui_sharedialogshare.h"
+#include "account.h"
+#include "json.h"
+#include "folderman.h"
+#include "folder.h"
+#include "accountmanager.h"
+#include "theme.h"
+#include "configfile.h"
+#include "capabilities.h"
+
+#include "thumbnailjob.h"
+#include "share.h"
+#include "sharee.h"
+
+#include "QProgressIndicator.h"
+#include <QBuffer>
+#include <QFileIconProvider>
+#include <QClipboard>
+#include <QFileInfo>
+#include <QCompleter>
+
+namespace OCC {
+
+ShareUserGroupDialog::ShareUserGroupDialog(AccountPtr account, const QString &sharePath, const QString &localPath, bool resharingAllowed, QWidget *parent) :
+   QDialog(parent),
+    _ui(new Ui::ShareUserGroupDialog),
+    _account(account),
+    _sharePath(sharePath),
+    _localPath(localPath),
+    _resharingAllowed(resharingAllowed)
+{
+    setAttribute(Qt::WA_DeleteOnClose);
+    setObjectName("SharingDialogUG"); // required as group for saveGeometry call
+
+    _ui->setupUi(this);
+
+    //Is this a file or folder?
+    _isFile = QFileInfo(localPath).isFile();
+
+    _ui->searchPushButton->setEnabled(false);
+    _ui->shareeView->hide();
+    _ui->searchMorePushButton->hide();
+    _ui->sharePushButton->hide();
+
+    _manager = new ShareManager(_account, this);
+    connect(_manager, SIGNAL(sharesFetched(QList<QSharedPointer<Share>>)), SLOT(slotSharesFetched(QList<QSharedPointer<Share>>)));
+    connect(_manager, SIGNAL(shareCreated(QSharedPointer<Share>)), SLOT(getShares()));
+}
+
+void ShareUserGroupDialog::done( int r ) {
+    ConfigFile cfg;
+    cfg.saveGeometry(this);
+    QDialog::done(r);
+}
+
+ShareUserGroupDialog::~ShareUserGroupDialog()
+{
+    delete _ui;
+}
+
+void ShareUserGroupDialog::on_shareeLineEdit_textEdited(const QString &text)
+{
+    if (text == "") {
+        _ui->searchPushButton->setEnabled(false);
+    } else {
+        _ui->searchPushButton->setEnabled(true);
+    }
+}
+
+void ShareUserGroupDialog::on_searchPushButton_clicked()
+{
+    ShareeModel *model = new ShareeModel(_account,
+                                         _ui->shareeLineEdit->text(),
+                                         _isFile ? QLatin1String("file") : QLatin1String("folder"),
+                                         _ui->shareeView);
+    _ui->shareeView->setModel(model);
+
+    _ui->shareeView->show();
+    _ui->searchMorePushButton->show();
+    _ui->sharePushButton->show();
+    _ui->sharePushButton->setEnabled(false);
+}
+
+void ShareUserGroupDialog::on_searchMorePushButton_clicked()
+{
+    //TODO IMPLEMENT
+}
+
+void ShareUserGroupDialog::on_shareeView_activated()
+{
+    _ui->sharePushButton->setEnabled(true);
+}
+
+void ShareUserGroupDialog::on_sharePushButton_clicked()
+{
+    const QModelIndex index = _ui->shareeView->currentIndex();
+
+    auto model = _ui->shareeView->model();
+
+    const QModelIndex shareWithIndex = model->index(index.row(), 2);
```

Comment:
O sure, but I think we need something else than the tableview for the final design. I just needed something I could properly check.